.github/workflows/system-tests-opensource.yml

```
@@ -55,165 +56,215 @@ on:
+  MLRUN_PYTHON_PACKAGE_INSTALLER: uv
 
 jobs:
-  run-system-tests-opensource-ci:
-    name: Run System Tests Open Source
+  prepare-inputs:
+    name: Prepare Open Source inputs
     runs-on: ubuntu-latest
+    outputs:
+      mlrun_docker_repo: ${{ steps.computed_params.outputs.mlrun_docker_repo }}
+      mlrun_docker_tag: ${{ steps.computed_params.outputs.mlrun_docker_tag }}
+      mlrun_ui_version: ${{ steps.computed_params.outputs.mlrun_ui_version }}
+      mlrun_docker_registry: ${{ steps.computed_params.outputs.mlrun_docker_registry }}
+      mlrun_system_tests_clean_resources: ${{ steps.computed_params.outputs.mlrun_system_tests_clean_resources }}
+      mlrun_version: ${{ steps.computed_params.outputs.mlrun_version }}
+      mlrun_version_specifier: ${{ steps.computed_params.outputs.mlrun_version_specifier }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Extract git hashes from upstream and latest version
+        id: git_upstream_info
+        run: |
+          # if the action is running on a PR, we need to get the latest commit hash from the PR branch
+          if [ -n "$PR_NUMBER" ]; then
+            echo "mlrun_hash=pr$PR_NUMBER" >> $GITHUB_OUTPUT
+          else
+            echo "mlrun_hash=$( \
+              cd /tmp && \
+              git clone --single-branch --branch development https://github.com/mlrun/mlrun.git mlrun-upstream 2> /dev/null && \
+              cd mlrun-upstream && \
+              git rev-list --until="1 hour ago" --max-count 1 --abbrev-commit HEAD && \
+              cd .. && \
+              rm -rf mlrun-upstream)" >> $GITHUB_OUTPUT
+          fi
+          
+          # NOTE: for now, always take the latest commit hash from the development branch
+          # TODO: take from relevant branch [pr head branch]
+          echo "ui_hash=$( \
+              cd /tmp && \
+              git clone --single-branch --branch development https://github.com/mlrun/ui.git mlrun-ui 2> /dev/null && \
+              cd mlrun-ui && \
+              git rev-list --until="1 hour ago" --max-count 1 --abbrev-commit HEAD && \
+              cd .. && \
+              rm -rf mlrun-ui)" >> $GITHUB_OUTPUT
+          echo "unstable_version_prefix=$(cat automation/version/unstable_version_prefix)" >> $GITHUB_OUTPUT
+        env:
+          PR_NUMBER: ${{ github.event.inputs.pr_number }}
+
+      - name: Set computed versions params
+        id: computed_params
+        run: |
+          export mlrun_hash=$(echo ${{ steps.git_upstream_info.outputs.mlrun_hash }})
+          export ui_hash=$(echo ${{ steps.git_upstream_info.outputs.ui_hash }})
+          echo "mlrun_hash=$mlrun_hash" >> $GITHUB_OUTPUT
+          echo "ui_hash=$ui_hash" >> $GITHUB_OUTPUT
+
+          echo "mlrun_version=$(echo ${{ steps.git_upstream_info.outputs.unstable_version_prefix }}+$mlrun_hash)" >> $GITHUB_OUTPUT
+          echo "mlrun_ui_version=${{ steps.git_upstream_info.outputs.unstable_version_prefix }}-$ui_hash" >> $GITHUB_OUTPUT
```

this might be needed
```suggestion
          echo "mlrun_ui_version=$(echo ${{ steps.git_upstream_info.outputs.unstable_version_prefix }}-$ui_hash)" >> $GITHUB_OUTPUT
```

---

it is actually not needed. the variables are templates without the need to echo-ed them